packages/datadog-plugin-amqplib/src/consumer.js

```diff
@@ -28,7 +28,13 @@ class AmqplibConsumerPlugin extends ConsumerPlugin {
       }
     })
 
-    if (this.config.dsmEnabled && message) {
+    if (
+      this.config.dsmEnabled &&
+      message &&
+      message.properties &&
+      message.properties.headers &&
+      message.properties.headers[CONTEXT_PROPAGATION_KEY]
```

Comment:
```suggestion
      message?.properties?.headers?.[CONTEXT_PROPAGATION_KEY]
```
We support no lower than Node 14, so we get [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) 🥳